bookops_worldcat/utils.py

```diff
@@ -26,13 +26,18 @@ def prep_oclc_number_str(oclcNumber: str) -> str:
         `oclcNumber` as str
 
     Raises:
-        InvalidOclcNumber: If `oclcNumber` argument is not a str or int.
+        InvalidOclcNumber: If `oclcNumber` argument does not match
+        OCLC number formatting rules.
     """
 
-    if oclcNumber.strip().startswith("ocm") or oclcNumber.strip().startswith("ocn"):
+    if re.match(r"^ocm[0-9]{,8}$", oclcNumber.strip()) or re.match(
```

Comment:
Initially, I was reluctant to see why regular experssions are needed (exta import, less readble that other string manipulation methods in Python), but with addition of lengh validation for a particular prefix that makes sense and the code would be more cumbersome without `re`. Good thinking!